internal/resolution/variable_sources/global_constraints/global_constraints.go

```diff
@@ -0,0 +1,101 @@
+package global_constraints
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/constraint"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/bundles_and_dependencies"
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+)
+
+type GlobalConstraintsVariable struct {
+	*input.SimpleVariable
+}
+
+func NewGlobalConstraintsVariable(id deppy.Identifier, atMostIDs ...deppy.Identifier) *GlobalConstraintsVariable {
+	return &GlobalConstraintsVariable{
+		SimpleVariable: input.NewSimpleVariable(id, constraint.AtMost(1, atMostIDs...)),
+	}
+}
+
+var _ input.VariableSource = &GlobalConstraintsVariableSource{}
+
+type GlobalConstraintsVariableSource struct {
+	inputVariableSource input.VariableSource
+}
+
+func NewGlobalConstraintsVariableSource(inputVariableSource input.VariableSource) *GlobalConstraintsVariableSource {
+	return &GlobalConstraintsVariableSource{
+		inputVariableSource: inputVariableSource,
+	}
+}
+
+func (g *GlobalConstraintsVariableSource) GetVariables(ctx context.Context, entitySource input.EntitySource) ([]deppy.Variable, error) {
+	variables, err := g.inputVariableSource.GetVariables(ctx, entitySource)
+	if err != nil {
+		return nil, err
+	}
+
+	// todo(perdasilva): better handle cases where a provided gvk is not found
+	//                   not all packages will necessarily export a CRD
+
+	pkgToBundleMap := map[string]map[deppy.Identifier]struct{}{}
+	gvkToBundleMap := map[string]map[deppy.Identifier]struct{}{}
+	for _, variable := range variables {
+		switch v := variable.(type) {
+		case *bundles_and_dependencies.BundleVariable:
+			bundleEntities := []*olmentity.BundleEntity{v.BundleEntity()}
+			bundleEntities = append(bundleEntities, v.Dependencies()...)
+			for _, bundleEntity := range bundleEntities {
+				// get bundleID package and update map
+				packageName, err := bundleEntity.PackageName()
+				if err != nil {
+					return nil, fmt.Errorf("error creating global constraints: %s", err)
```

Comment:
wrap `err` via `%w`, here and below

Reply:
I've done a project search and hopefully addressed all of them (I'll check the next branches too)

Reply:
Odd... this is still showing up as `%s`?

Reply:
Other than this, looks fine.

Reply:
I see this too.

Reply:
hmmmm - maybe I screwed something up - hold on (sorry for this!)

Reply:
fixed